hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -865,6 +906,20 @@ public synchronized X509Certificate getRootCACertificate() {
     return null;
   }
 
+  @Override
+  public Set<X509Certificate> getAllRootCaCerts() {
+    return rootCaCertificates.stream().
+        map(this::firstCertificateFrom)
```

Comment:
Is there a use case where we use any secondary certificate from CA certificate files? Also is there a case when we have a CA certificate file with more than one certificate? (I believe this we might have but just after SCM decommission, but correct me if I am wrong.)
If we do not have, we can store the first certificate only in memory, and then we do not need this mapping all the time, and we can return an unmodifiableSet based on the stored certs simply. What do you think?

Reply:
Thanks, that's a good idea. I'm going to double check if these files are really only containing only one cert and implement your suggested change after.